src/scripts/main.js

```diff
@@ -1,11 +1,29 @@
 'use strict';
 
+const body = document.querySelector('body');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (!element) {
+      const error = `It was NOT ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
```

Comment:
There is a potential runtime error here. If `element` is null or undefined, accessing `element.nodeName` or `element.id` will throw an error. Consider checking if `element` is truthy before accessing its properties.